src/DotLiquid/StandardFilters.cs

```diff
@@ -751,12 +751,19 @@ public static object Last(IEnumerable array)
         /// <param name="operand">Number to be added to input</param>
         public static object Plus(Context context, object input, object operand)
         {
-            if (context.SyntaxCompatibilityLevel >= SyntaxCompatibility.DotLiquid21)
-                return DoMathsOperation(context, input, operand, Expression.AddChecked);
+            try
+            {
+                if (context.SyntaxCompatibilityLevel >= SyntaxCompatibility.DotLiquid21)
+                    return DoMathsOperation(context, input, operand, Expression.AddChecked);
 
-            return input is string
-                ? string.Concat(input, operand)
-                : DoMathsOperation(context, input, operand, Expression.AddChecked);
+                return input is string
+                    ? string.Concat(input, operand)
+                    : DoMathsOperation(context, input, operand, Expression.AddChecked);
+            }
+            catch (OverflowException) when (input is int)

```

Comment:
wouldn't uint also have a boundary? and smaller integer values, like short? Additionally, Ruby implementation supports adding 1 to decimal max without losing precision.

Reply:
I tried bytes and short, they worked fine without modification, I can add a few tests to demonstrate it. I didn't try uint.

Reply:
BigInteger is out of scope for a bug fix, that support would be a feature in itself.